src/util/io.c

```diff
@@ -146,27 +152,41 @@ socket_connect (
     uint16_t port,
     SOCKET *sock)
 {
-    struct sockaddr_in sockaddr = { 0 };
+    static const struct addrinfo hints = { .ai_socktype = SOCK_STREAM,
+        .ai_family = AF_UNSPEC, .ai_protocol = IPPROTO_TCP,
+        .ai_flags=AI_CANONIDN };
+    struct addrinfo *retp = NULL;
+    struct addrinfo *pprev = NULL;
+    struct addrinfo *p;
+    char port_str[MAX_PORT_STR_LEN];
```

Comment:
char port_str[MAX_PORT_STR_LEN + 1]; sizeof doesn't count the terminating NULL.

Reply:
Yes it does count the terminating NUL character.  You may be confusing sizeof() with strlen().  Here's an example:

#include <stdio.h>
#include <string.h>
void main() {
        printf("strlen of \"foo\" is %lu\n", strlen("foo"));
        printf("sizeof of \"foo\" is %lu\n", sizeof("foo"));
}
(glacier ~/tmp) ./foo
strlen of "foo" is 3
sizeof of "foo" is 4